### PR TITLE
fix edge cases for `databricks_permissions` resource

### LINKED ISSUE #2087: [ISSUE] Restricting password usage to admins is denied in `databricks_permissions`
We have SSO on our workspaces and intend to restrict password usage to only admins, as a way to log in should SSO stop working, in order to resolve the issue. When attempting to set this via Terraform, plans fail with ``Error: It is not possible to restrict any permissions from `admins` ``

### Configuration
```hcl
resource "databricks_permissions" "password_usage" {
  authorization = "passwords"

  access_control {
    group_name       = "admins"
    permission_level = "CAN_USE"
  }
}
```

### Expected Behavior

The password usage policy should allow setting `CAN_USE` for the admins group. This is possible if done via the Databricks console

image

### Actual Behavior
```
 Error: It is not possible to restrict any permissions from `admins`.
│ 
│   with module.workspace_sandbox.databricks_permissions.password_usage,
│   on modules/workspace/permissions.tf line 27, in resource "databricks_permissions" "password_usage":
│   27: resource "databricks_permissions" "password_usage" {
│ 
```

### Debug Output
```
[WARN]  provider.terraform-provider-databricks_v1.11.1: Truncating attribute path of 1 diagnostics for TypeSet: timestamp=2023-03-08T12:25:50.547-0500
2023-03-08T12:25:50.547-0500 [ERROR] provider.terraform-provider-databricks_v1.11.1: Response contains error diagnostic: tf_rpc=ValidateResourceTypeConfig @caller=/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/diag/diagnostics.go:55 @module=sdk.proto diagnostic_attribute=AttributeName("access_control") diagnostic_detail= diagnostic_severity=ERROR tf_resource_type=databricks_permissions diagnostic_summary="It is not possible to restrict any permissions from `admins`." tf_proto_version=5.3 tf_provider_addr=registry.terraform.io/databricks/databricks tf_req_id=0696e93e-9ad4-40c3-8591-9969f2e9bf13 timestamp=2023-03-08T12:25:50.547-0500
2023-03-08T12:25:50.547-0500 [WARN]  provider.terraform-provider-databricks_v1.11.1: Truncating attribute path of 1 diagnostics for TypeSet: timestamp=2023-03-08T12:25:50.547-0500
2023-03-08T12:25:50.547-0500 [WARN]  provider.terraform-provider-databricks_v1.11.1: Truncating attribute path of 1 diagnostics for TypeSet: timestamp=2023-03-08T12:25:50.547-0500
2023-03-08T12:25:50.547-0500 [ERROR] provider.terraform-provider-databricks_v1.11.1: Response contains error diagnostic: diagnostic_attribute=AttributeName("access_control") diagnostic_detail= diagnostic_summary="It is not possible to restrict any permissions from `admins`." tf_req_id=4aaf4edc-bdbf-c6d9-a1ab-e0c29508a361 @caller=/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/diag/diagnostics.go:55 @module=sdk.proto diagnostic_severity=ERROR tf_proto_version=5.3 tf_rpc=ValidateResourceTypeConfig tf_provider_addr=registry.terraform.io/databricks/databricks tf_resource_type=databricks_permissions timestamp=2023-03-08T12:25:50.547-0500
2023-03-08T12:25:50.547-0500 [ERROR] vertex "module.workspace_sandbox.databricks_permissions.password_usage" error: It is not possible to restrict any permissions from `admins`.
2023-03-08T12:25:50.547-0500 [ERROR] vertex "module.workspace_acm.databricks_permissions.password_usage" error: It is not possible to restrict any permissions from `admins`.
2023-03-08T12:25:50.547-0500 [ERROR] provider.terraform-provider-databricks_v1.11.1: Response contains error diagnostic: diagnostic_severity=ERROR tf_proto_version=5.3 tf_resource_type=databricks_permissions diagnostic_attribute=AttributeName("access_control") tf_provider_addr=registry.terraform.io/databricks/databricks @module=sdk.proto tf_req_id=efc9a692-03b0-0227-3113-eecb12f2f1bd tf_rpc=ValidateResourceTypeConfig @caller=/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/diag/diagnostics.go:55 diagnostic_detail= diagnostic_summary="It is not possible to restrict any permissions from `admins`." timestamp=2023-03-08T12:25:50.547-0500
2023-03-08T12:25:50.547-0500 [ERROR] vertex "module.workspace_core.databricks_permissions.password_usage" error: It is not possible to restrict any permissions from `admins`.
2023-03-08T12:25:50.592-0500 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/databricks/databricks/1.11.1/darwin_arm64/terraform-provider-databricks_v1.11.1 args=[.terraform/providers/registry.terraform.io/databricks/databricks/1.11.1/darwin_arm64/terraform-provider-databricks_v1.11.1]
```

### Terraform and provider versions
```
Terraform v1.3.8
on darwin_arm64
+ provider registry.terraform.io/alxrem/jsonnet v2.1.0
+ provider registry.terraform.io/databricks/databricks v1.9.2
+ provider registry.terraform.io/hashicorp/aws v4.50.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
```

### LINKED ISSUE #2134: [ISSUE] Cannot update permissions for `workspace`: `Cannot remove admins's CAN_MANAGE permission`
Hi team,

An issue occurs when modifying the permissions on the root level of the workspace.

### Configuration
```hcl
resource "databricks_permissions" "root_workspace_permissions" {
  directory_id = "0" # Root folder of the databricks workspace
  access_control {
    group_name       = "AAD_SDP_Data_Engineers"
    permission_level = "CAN_READ"
  }

  access_control {
    group_name       = "AAD_SDP_Data_Operators"
    permission_level = "CAN_READ"
  }
}
```

### Expected Behavior
When running `terraform apply`, the permissions should set.

### Actual Behavior
`terraform plan` runs successfully. However, `terraform apply` gives the following error
```
│ Error: cannot update permissions: Cannot remove admins's CAN_MANAGE permission on 0
│ 
│   with databricks_permissions.root_workspace_permissions,
│   on workspace.tf line 35, in resource "databricks_permissions" "root_workspace_permissions":
│   35: resource "databricks_permissions" "root_workspace_permissions" {
│ 
╵
```

### Steps to Reproduce
1. Run `terraform init`
2. Run `terraform plan` (which will run successfully
3. Run `terraform apply`  

### Terraform and provider versions
```
Terraform v1.3.5
on windows_amd64
+ provider registry.terraform.io/databricks/databricks v1.13.0
+ provider registry.terraform.io/hashicorp/azuread v2.33.0
+ provider registry.terraform.io/hashicorp/azurerm v3.43.0
```

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes
Fix edge cases when setting permissions for root directory and password/token usage

Closes #2087
closes #2134

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

**Comment by nkvuong** [comment]
acceptance tests passed on all 3 clouds

**Review by nfx — CHANGES_REQUESTED** [review]

File: permissions/resource_permissions.go [hunk 1]
```diff
@@ -359,23 +357,6 @@ func ResourcePermissions() *schema.Resource {
...
 		}
 		s["access_control"].MinItems = 1
-		if groupNameSchema, err := common.SchemaPath(s,
-			"access_control", "group_name"); err == nil {
-			groupNameSchema.ValidateDiagFunc = func(i any, p cty.Path) diag.Diagnostics {
-				if v, ok := i.(string); ok {
-					if strings.ToLower(v) == "admins" {
-						return diag.Diagnostics{
-							{
-								Summary:       "It is not possible to restrict any permissions from `admins`.",
```
[the remark this answers is not in the document]
Comment (nkvuong) [reply permissions/resource_permissions.go:369]: I moved this logic down to Create (same place where we validate decreasing permission for current users), as it is possible to set permission on authorization/passwords for admins #2087

**Review by nfx — APPROVED** [review]
